File: frontend/lib/data/api/api_paths.dart
/// Centralized API path constants mirroring the backend's documented routes
/// (`docs/api-overview.md`).
///
/// Feature endpoints are composed on top of [v1]; this keeps the version
/// prefix and the contract in one place.
abstract final class ApiPaths {
  /// Versioned base path of the REST API.
  static const String v1 = '/api/v1';

  /// Authentication endpoints.
  static const String authLogin = '$v1/auth/login';
  static const String authRegister = '$v1/auth/register';
  static const String authRefresh = '$v1/auth/refresh';
  static const String authLogout = '$v1/auth/logout';
  static const String authMe = '$v1/auth/me';

  /// Profile endpoints.
  static const String profile = '$v1/profile';

  /// Dashboard (owner-scoped summary counts) and task-list endpoints.
  static const String dashboard = '$v1/dashboard';
  static const String tasks = '$v1/tasks';
}

File: backend/src/main/java/com/todoapp/controller/ApiPaths.java
package com.todoapp.controller;

/**
 * Central API path constants.
 * <p>
 * Every controller maps under {@link #V1} to keep versioning consistent.
 */
public final class ApiPaths {

    public static final String V1 = "/api/v1";
    public static final String AUTH = V1 + "/auth";
    public static final String PROFILE = V1 + "/profile";
    public static final String TASKS = V1 + "/tasks";
    public static final String DASHBOARD = V1 + "/dashboard";
    public static final String HEALTH = V1 + "/health";

    private ApiPaths() {
    }
}